espnet2/train/epoch_iter_factory.py

```diff
@@ -0,0 +1,96 @@
+from abc import ABC
+from abc import abstractmethod
+from typing import Any
+from typing import Sequence
+
+import numpy as np
+from torch.utils.data.dataloader import DataLoader
+
+
+class AbsIterFactory(ABC):
+    @abstractmethod
+    def build_iter(self, epoch: int, shuffle: bool = None) -> DataLoader:
+        raise NotImplementedError
+
+
+class EpochIterFactory(AbsIterFactory):
+    def __init__(
+        self,
+        dataset,
+        batches: Sequence[Sequence[Any]],
+        num_iters_per_epoch: int = None,
```

Comment:
If the len(batches) = 4, 1 epoch w/ num_iters_per_epoch = None and 4 epochs w/ num_iters_per_epoch = 1 will be exactly the same?

Reply:
Yes if the implementation is correct.